src/runtime/cuda/cuda_device_api.cc

```diff
@@ -241,5 +242,28 @@ TVM_REGISTER_GLOBAL("device_api.cpu_pinned").set_body([](TVMArgs args, TVMRetVal
   *rv = static_cast<void*>(ptr);
 });
 
+TVM_REGISTER_GLOBAL("profiling.timer.gpu").set_body_typed([](TVMContext ctx) {
+  cudaEvent_t start;
+  cudaEvent_t stop;
+  cudaEventCreate(&start);
+  cudaEventCreate(&stop);
+  cudaEventRecord(start, CUDAThreadEntry::ThreadLocal()->stream);
+  return TypedPackedFunc<TypedPackedFunc<int64_t()>()>(
+      [=]() {
+        cudaEventRecord(stop, CUDAThreadEntry::ThreadLocal()->stream);
+        return TypedPackedFunc<int64_t()>(
+            [=]() -> int64_t {
+              cudaEventSynchronize(stop);
+              float milliseconds = 0;
```

Comment:
how does it work accessing `stop` from here when its copy was written to above? how might this work as a general mechanism? this does imply that starting and stopping a timer may load the dynamic memory allocator, which is bad for general timing API.

Reply:
`stop` is an opaque type copied from above. I am unsure if the allocator will be called. If so, it will be for allocating the closure.

Reply:
it matters a lot when the allocator is called relative to the timing calls, and being able to understand that just from reading the code is quite valuable

Reply:
I guess chatting further--I think we already generally are not timing right inside the kernel functions. at a later time if we start doing that, i'd like us to consider things at this level, but we don't need to alter the whole approach. we may be able to help though by moving the std::function instantiation to the Start() call